src/lksearch/CatalogSearch.py

```diff
@@ -0,0 +1,616 @@
+) -> SkyCoord:
+    """
+    Convert a table input to astropy.coordinates.SkyCoord object
+
+    Parameters
+    ----------
+    table : astropy.table.Table
+        astropy.table.Table which contains the coordinates of targets and proper motion values
+    equinox: astropy.time.Time
+        The equinox for the catalog
+    epoch : astropy.time.Time
+        Desired time of the observation
+
+    Returns
+    -------
+    coords : astropy.coordinates.SkyCoord
+       SkyCoord object with RA, Dec, equinox, and proper motion parameters.
+    """
+
+    if equinox is None and catalog is None:
+        _, catalog = _parse_id(table[0][0])
+    if equinox is None and catalog is not None:
+        equinox = _Catalog_Dictionary[catalog]["equinox"]
+    if epoch is None and catalog is not None:
+        epoch = equinox
+
+    # We need to remove any nan values from our proper  motion list
+    # Doing this will allow objects which do not have proper motion to still be displayed
+    if "pmRA" in table.keys():
+        table["pmRA"] = np.ma.filled(table["pmRA"].astype(float), 0.0)
+    else:
+        table["pmRA"] = 0.0 * u.mas / u.yr
+
+    if "pmDE" in table.keys():
+        table["pmDE"] = np.ma.filled(table["pmDE"].astype(float), 0.0)
+    else:
+        table["pmDE"] = 0.0 * u.mas / u.yr
+
+    # If an object does not have a parallax then we treat it as if the object is an "infinite distance"
+    # and set the parallax to 1e-7 arcseconds or 10Mpc.
+    if "Plx" in table.keys():
+        table["Plx"] = np.ma.filled(table["Plx"].astype(float), 1e-4)
+    else:
+        table["Plx"] = 1e-4 * u.mas
+
+    if "RAJ2000" in table.keys() and "DEJ2000" in table.keys():
+        ra = table["RAJ2000"]
+        dec = table["DEJ2000"]
+    elif "RA" in table.keys() and "Dec" in table.keys():
+        ra = table["RA"] * u.deg
+        dec = table["Dec"] * u.deg
+    else:
+        raise KeyError("RA, DEC Disambiguation failed")
+    # Suppress warning caused by Astropy as noted in issue 111747 (https://github.com/astropy/astropy/issues/11747)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="negative parallaxes")
+
+        # Get the input data from the table
+        c = SkyCoord(
+            ra=ra,
+            dec=dec,
+            distance=Distance(parallax=table["Plx"].quantity, allow_negative=True),
+            pm_ra_cosdec=table["pmRA"],
+            pm_dec=table["pmDE"],
+            frame="icrs",
+            obstime=equinox,
+        )
+
+    # Suppress warning caused by Astropy as noted in issue 111747 (https://github.com/astropy/astropy/issues/11747)
+    if epoch != equinox:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="ERFA function")
+            warnings.filterwarnings("ignore", message="invalid value")
+            c = c.apply_space_motion(new_obstime=epoch)
+    return c
+
+
+class CatalogResult(Table):
```

Comment:
If we remove this, we can remove the functionality to turn a table into a sky coordinate, and then everything can be `pandas` dataframes. I think this is the best trade,  (we still have the functionality with return_skycoord=True) and we get to use pandas throughout the `lksearch` package outputs.